src/orion/algo/robo/base.py

```diff
@@ -354,73 +394,57 @@ def suggest(self, num=None):
 
             if not candidates:
                 break
-
+        assert all(isinstance(sample, Trial) for sample in samples), samples
         return samples
 
-    def _suggest(self, num, function):
-        points = []
-
+    def _suggest(
+        self, num: int, function: Callable[[int], Iterable[Trial]]
+    ) -> list[Trial]:
+        trials: list[Trial] = []
         attempts = 0
         max_attempts = 100
-        while len(points) < num and attempts < max_attempts:
-            for candidate in function(num - len(points)):
+        while len(trials) < num and attempts < max_attempts:
+            for candidate in function(num - len(trials)):
                 if not self.has_suggested(candidate):
                     self.register(candidate)
-                    points.append(candidate)
+                    trials.append(candidate)
 
                 if self.is_done:
-                    return points
+                    return trials
 
             attempts += 1
-            print(attempts)
-
-        return points
+        return trials
 
-    def _suggest_random(self, num):
-        def sample(num):
+    def _suggest_random(self, num: int) -> list[Trial]:
+        def sample(num: int) -> list[Trial]:
             return self.space.sample(
                 num, seed=tuple(self.rng.randint(0, 1000000, size=3))
             )
 
         return self._suggest(num, sample)
 
-    def _suggest_bo(self, num):
+    def _suggest_bo(self, num: int) -> list[Trial]:
         # pylint: disable = unused-argument
-        def suggest_bo(num):
+        def suggest_bo(num: int) -> list[Trial]:
             # pylint: disable = protected-access
-            point = list(self.robo.choose_next(self.X, self.y))
-
-            # If already suggested, give corresponding result to BO to sample another point
-            if self.has_suggested(point):
-                result = self._trials_info[self.get_id(point)][1]
-                if result is None:
-                    results = []
-                    for _, other_result in self._trials_info.values():
-                        if other_result is not None:
-                            results.append(other_result["objective"])
-                    result = {"objective": numpy.array(results).mean()}
-
-                self._bo_duplicates.append((point, result))
-
-                # self.optimizer.tell([point], [result])
-                return []
-
-            return [point]
+            assert self.robo is not None
+            X, y = self.XY
+            point = list(self.robo.choose_next(X, y))
+            trial = tuple_to_trial(point, self.space)
+            return [trial]
+            # TODO: Double-check, but I don't think this kind of thing is needed anymore.
```

Comment:
Yeah, I think this is not needed anymore now that the Wrapper will try algo.suggest many times if it returns many times trials that maps to the same one.

Reply:
Fixed in f0d6e251397b40ec7ff6bfcda8579d6ae813f484